src/apiController.js

```diff
@@ -0,0 +1,217 @@
+let axios = require('axios');
+let config = require('/src/config.js');
+
+
+axios.defaults.baseURL = config.server;
+axios.defaults.headers.common['Authorization'] = config.token;
+
+//I've basically created functions for each type of call. Didn't actually have them do anything yet because I haven't fully figured out how to use these.
+// They'll also need to have actual stuff written in them to actually use them, but I figure people can fill in the info for their own functions as we go on.
+let apiController = {
+
+
+  // Products API calls
+  getProduct: (pages, countNum) => {
+    let page = pages || 1;
+    let count = countNum || 5;
+    let params = {page: page, count: count};
+    axios.get('/products', {params})
+    .then((data) => {
+      console.log(data);
+    })
+    .catch((err) => {
+      console.error('Oh noes ', err);
+    })
+  },
+  //I'm unsure of whether I'm doing this the right way since I'm not sure how exactly the productId is going to be passed on, but I assume something like this.
+  getProductById: (productId) => {
+    axios.get(`/product/${productId}`)
+    .then((data) => {
+      console.log(data);
+    })
+    .catch((err) => {
+      console.error('Oh noes ', err);
+    })
+  },
+
+  getProductStyles: (productId) => {
+    axios.get(`/product/${productId}/styles`)
+    .then((data) => {
+      console.log(data);
+    })
+    .catch((err) => {
+      console.error('Oh noes ', err);
+    })
+  },
+
+  getProductRelated: (productId) => {
+    axios.get(`/product/${productId}/related`)
+    .then((data) => {
+      console.log(data);
+    })
+    .catch((err) => {
+      console.error('Oh noes ', err);
+    })
+  },
+  //Reviews API Calls
+  getReviews: (pages, countNum, sortVal, productId) => {
```

Comment:
Like, this one is a good example of how I think the object destructuring would help. The only parameter that isn't optional here is productId, so I would have it look like `getReviews: (productId, {pages, countNum,  sortVal}) => {`, basically all optional params should be last, and in an "options" object